storm-core/test/jvm/org/apache/storm/LocalStateTest.java

```diff
@@ -0,0 +1,54 @@
+package org.apache.storm;
+
+import org.apache.storm.generated.GlobalStreamId;
+import org.apache.storm.testing.TmpPath;
+import org.apache.storm.utils.LocalState;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class LocalStateTest {
+
+    @Test
+    public void testLocalState() throws IOException{
+        TmpPath dir1_tmp = new TmpPath();
+        TmpPath dir2_tmp = new TmpPath();
+        GlobalStreamId globalStreamId_a = new GlobalStreamId("a","a");
+        GlobalStreamId globalStreamId_b = new GlobalStreamId("b","b");
+        GlobalStreamId globalStreamId_c = new GlobalStreamId("c","c");
+        GlobalStreamId globalStreamId_d = new GlobalStreamId("d","d");
+
+        LocalState ls1 = new LocalState(dir1_tmp.getPath());
+        LocalState ls2 = new LocalState(dir2_tmp.getPath());
+        Assert.assertTrue(ls1.snapshot().isEmpty());
+
+        ls1.put("a",globalStreamId_a);
+        ls1.put("b",globalStreamId_b);
+        Assert.assertTrue(ls1.snapshot().containsKey("a"));
```

Comment:
Below comparisons are broken: `{"a" globalStreamId_b "b" globalStreamId_a}` will pass.

I guess you can compare two Maps directly since Map.equals defines how implementation should compare each other properly.
https://docs.oracle.com/javase/7/docs/api/java/util/Map.html#equals(java.lang.Object)